bears/python/PyImportSortBear.py

```diff
@@ -176,13 +209,38 @@ def run(self, filename, file,
             isort_settings['known_standard_library'] = (
                 known_standard_library_imports)
 
-        sort_imports = SortImports(file_contents=''.join(file),
-                                   **isort_settings)
-        new_file = tuple(sort_imports.output.splitlines(True))
+        if treat_seperated_imports_independently:
+            import_stmts = PyImportSortBear.seperate_imports(file)
+            sorted_imps = []
+            for units in import_stmts:
+                sort_imports = SortImports(file_contents=''.
+                                           join([x[1] for x in units]),
+                                           **isort_settings)
+                sort_imports = sort_imports.output.splitlines(True)
+                sorted_imps.append((units, sort_imports))
+
+            diff = Diff(file)
+            for old, new in sorted_imps:
+                start = old[0][0]
+                end = start + len(old) - 1
+                diff.delete_lines(start, end)
+                assert isinstance(new, list)
+                diff.add_lines(start, list(new))
+
+            if diff.modified != diff._file:
+                yield Result(self,
```

Comment:
this `Result` being yielded is identical to the existing `Result` being yielded.  The code needs to be adjusted so there is only one `Result` being yielded.